api/client/client.go

```diff
@@ -370,10 +370,24 @@ func (c *Client) dialGRPC(ctx context.Context, addr string) error {
 		grpc.WithChainUnaryInterceptor(
 			otelgrpc.UnaryClientInterceptor(),
 			metadata.UnaryClientInterceptor,
+			func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+				value := ctx.Value(utils.ContextClientAddr)
+				if addr, ok := value.(*net.TCPAddr); ok {
+					ctx = metadata.AddMetadataToContext(ctx, map[string]string{"client_addr": addr.String()})
```

Comment:
Using the metadata seems to be along the right lines here, but I'm not sure this is secure yet.

Hypothetically, an external gRPC client could also set this metadata, and spoof its IP address to circumvent the IP pinning of certificates. We need to establish that this comes from a trusted source (e.g the Proxy). My gut feeling on this is some kind of signing. Either we need an established pre-shared key between the Proxy and the Auth server, or the Proxy needs to use its public/private key to sign it, and we also need to include the Proxy's certificate in order to validate that its been signed by something with the Proxy role.